ethereum/eip712/eip712.go

```diff
@@ -363,6 +355,55 @@ func jsonNameFromTag(tag reflect.StructTag) string {
 	return parts[0]
 }
 
+func unpackIfAny(
+	cdc codectypes.AnyUnpacker,
+	_fieldType reflect.Type,
+	_field reflect.Value,
+	assumeMsgStructure bool,
```

Comment:
not sure what this bool is for, I'd consider splitting this function into 2 different ones